ext/liquid_c/vm.c

```diff
@@ -3,67 +3,383 @@
 #include "liquid.h"
 #include "vm.h"
 #include "resource_limits.h"
+#include "context.h"
 
 ID id_render_node;
 ID id_ivar_interrupts;
 ID id_ivar_resource_limits;
+ID id_to_s;
+ID id_vm;
+ID id_strainer;
+ID id_filter_methods_hash;
+ID id_strict_filters;
+ID id_global_filter;
 
-void liquid_vm_next_instruction(const uint8_t **ip_ptr, const size_t **const_ptr_ptr)
+static VALUE cLiquidCVM;
+
+typedef struct vm {
+    c_buffer_t stack;
+    VALUE strainer;
+    VALUE filter_methods;
+    VALUE interrupts;
+    VALUE resource_limits_obj;
+    resource_limits_t *resource_limits;
+    VALUE global_filter;
+    bool strict_filters;
+} vm_t;
+
+static void vm_mark(void *ptr)
```

Comment:
This doesn't mark `global_filter`. I think this should be fine since the context should be holding the reference. This could lead to bugs if the context's `global_filter` gets updated.

Reply:
Thanks for catching that.  That is indeed a bug, which can also result in problems from GC compaction, which will move objects that aren't marked from C code.